Comando size + refactors

PierreVieira opened the pull request:
## O que esse PR faz?
- Adição de pastas de IDE no .gitignore
- Criação de um package utils
- Remoção de código duplicado
- Comando `.size`

PierreVieira committed:
[+] feature size

PierreVieira committed:
fix

Agronault commented on .gitignore:
```diff
@@ -10,3 +10,7 @@ node_modules
 # end managed by open-wa
+
+# IDE's
+.vscode
```
n coloca pra ignorar n, é lá q configura o debugger pro vscode, isso é útil

Eliabe45 replied:
Pior que nem fui eu, provavelmente foi o commit do @julioCROS que eu deixei passar hahaha

Agronault commented on .vscode/launch.json:
```diff
@@ -1,18 +0,0 @@
-{
```
deleta n pô

Eliabe45 replied:
Num deletei :sob: , eu uso tbm hahauau

Agronault commented on src/bot/commands/command-list/size.ts:
```diff
@@ -0,0 +1,23 @@
+  // hidden: true,
+};
+
+export default size
```
faltou blank line eof

eliabefranca replied:
Tinha que criar um .eslintrc e deixar configurado no projeto junto com o prettier, aí bota uns hooks do git pra formatar com o lint-staged, mas mó preguiça. Qqr dia eu faço isso :+1:

Agronault replied:
> Tinha que criar um .eslintrc e deixar configurado no projeto junto com o prettier, aí bota uns hooks do git pra formatar com o lint-staged, mas mó preguiça. Qqr dia eu faço isso 👍

@Eliabe45 vou colocar na priorização aqui, boa xD

Agronault commented on src/bot/commands/command-list/utils/getContactNumber.ts:
```diff
@@ -0,0 +1,15 @@
+  return firstMember.id.split('@')[0];
+}
+
+export default getContactNumber
```
faltou blank line eof

Agronault commented on src/bot/commands/command-list/utils/getPersonExists.ts:
```diff
@@ -0,0 +1,9 @@
+  );
+}
+
+export default getPersonExists
```
faltou blank line eof

Agronault commented on src/bot/commands/command-list/utils/getRandomInterval.ts:
```diff
@@ -0,0 +1,5 @@
+  return Math.floor(Math.random() * (max - min + 1) + min)
+}
+
+export default getRandomInterval
```
blank line eof

Agronault commented on src/bot/commands/command-list/utils/outputErrorMessage.ts:
```diff
@@ -0,0 +1,11 @@
+  );
+}
+
+export default outputErrorMessage
```
blank line eof

Agronault commented on src/bot/commands/command-list/utils/validPerson.ts:
```diff
@@ -0,0 +1,20 @@
+}
+
+
+export default validPerson
```
blank line eof

Agronault commented on src/bot/commands/command-list/kiss.ts:
```diff
@@ -1,56 +1,36 @@
-    );
-    return;
+  const kissedPerson = await validPerson(client, message, value, 'Você precisa me dizer qual pessoa você quer beijar')
+  if (kissedPerson != null) {
```
```suggestion
  if (kissedPerson) {
```

Agronault commented on src/bot/commands/command-list/rank.ts:
```diff
@@ -9,7 +9,7 @@ const func: Command = async ({ client, message }) => {
 
   if (rank.length === 0) {
-    client.reply(
+    await client.reply(
```
não precisa dar await aqui, esse é o fim do fluxo

Agronault commented on src/bot/commands/command-list/size.ts:
```diff
@@ -0,0 +1,23 @@
+const func: Command = async ({client, message, value}) => {
+  const personToCalculateSize = await validPerson(client, message, value, 'Você precisa me informar o nome de quem quer saber o tamanho do piupiu')
+  const sizePiupiu = getRandomInterval(2, 20)
+  const piupiu = '8' + '='.repeat(sizePiupiu) + 'D'
```
acho q é melhor fazer o size/3, pq 10cm ser um pinto "8==========D" não faz sentido

PierreVieira replied:
era pra ser
```.ts
 const piupiu = '8' + '='.repeat(sizePiupiu - 2) + 'D'
```
Mas esqueci do -2 na hora hihi.

Agronault replied:
oq acha de dividir por 3?
5cm -> 8=D
10cm -> 8===D
20cm -> 8======D

Agronault commented on src/bot/commands/command-list/size.ts:
```diff
@@ -0,0 +1,23 @@
+
+const func: Command = async ({client, message, value}) => {
+  const personToCalculateSize = await validPerson(client, message, value, 'Você precisa me informar o nome de quem quer saber o tamanho do piupiu')
+  const sizePiupiu = getRandomInterval(2, 20)
```
vc abandonou aquela ideia de usar o user como seed pra fazer o valor sempre ser o mesmo? Vc pode somar o valor do ASCII dos chars do nome do usuário e fazer o resto da divisão disso por 19 dps somar 2

PierreVieira replied:
Abandonei não, mas seria só uma v1 do `.size`. Eu entendi a lógica, mas como o pr tinha mais modificações (muita extração de código) achei necessário dar esse merge primeiro para evitar conflitos futuros.

Agronault replied:
> Abandonei não, mas seria só uma v1 do `.size`. Eu entendi a lógica, mas como o pr tinha mais modificações (muita extração de código) achei necessário dar esse merge primeiro para evitar conflitos futuros.

Show, boa, vou colocar isso no backlog ent

Agronault reviewed (changes requested):
deixei algumas sugestões

eliabefranca committed:
Update src/bot/commands/command-list/kiss.ts

Co-authored-by: Alexandre Martins <31781535+Agronault@users.noreply.github.com>

eliabefranca reviewed (changes requested):
Opa, valeu pela contribuição :heart: 

Dá uma olhadinha nesses pontos:


1. Agora os comandos têm categoria, já tá tipado, no caso esse comando é da categoria 'funny', inclui isso no export do comando por favor
2. Tá dando um conflitinho com a main no arquivo do comando .kiss, dá um reset aí no seu arquivo e manda um git pull
2. Consegue exportar o getRandomContactNumber e o validPerson com named export ao invés de default export? Só pra deixar no padrão mesmo
3. N precisa botar o .vscode dentro do .gitignore

PierreVieira committed:
fix + refactors

PierreVieira committed:
Merge remote-tracking branch 'origin/feature/size' into feature/size

PierreVieira committed:
insert blank line in launch.json

eliabefranca committed:
Update virgindade.ts

@PierreVieira  só remover esses espaços aqui, pq eles são computados na hr do print e quebram a msg

eliabefranca reviewed (approved):
Perfeito :heart: 

Excelente contribuição, melhorou muito a legibilidade :smile:
